tests/app/test_jinja_filters.py

```diff
@@ -95,6 +96,71 @@ def test_format_date(self):
 
         self.assertEqual(format_value, '1 January 2017')
 
+    def test_format_conditional_date_notvalid(self):
+        # Given
+
+        error_list = [(1, '1'),
+                      (None, 1),
+                      (1, None),
+                      (1, ''),
+                      ('', 9),
+                      (1, 3)]
+
+        for double in error_list:
+            date = double[0]
+            date2 = double[1]
+        with self.assertRaises(Exception) as exception:
```

Comment:
I think this indentation is incorrect so its only currently testing the last value in the list?